File: backend/src/main/java/com/github/kalheeso/provax/domain/Agenda.java
package com.github.kalheeso.provax.domain;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Setter;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@Entity
public class Agenda implements Serializable {
    @Setter(onMethod_ = @Deprecated)
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    // Data e Hora do agendamento
    private LocalDateTime dataHora;
    private Situacao situacao;
    private String dataSituacao;
    private String obervacoes;

    @ManyToOne
    @JoinColumn(nullable = false, name = "vacina_id")
    private Vacina vacina;

    @ManyToOne
    @JoinColumn(nullable = false, name = "usuario_id")
    private Usuario usuario;
}

File: frontend/src/app/vacine/core/entities/Agenda.ts
import Vacina from "./Vacina";


export default class Agenda {
  id !: number
  data !: Date
  situacao !: string

  vacina !: Vacina
  paciente !: any

  constructor(id:number, data : Date, situacao : string, vacina : Vacina, paciente : any){
    this.id = id
    this.data = data
    this.situacao = situacao
    this.vacina = vacina
    this.paciente = paciente
  }
}
